src/twisted/trial/test/detests.py

```diff
@@ -208,6 +215,32 @@ def timedOut(err):
 
     test_errorPropagation.timeout = 0.1  # type: ignore[attr-defined]
 
+    async def test_cancelSuppression(self):
+        async def sleep(delay):
+            await task.deferLater(clock=reactor, delay=delay)
+
+        try:
+            await sleep(0.2)
+        except defer.CancelledError:
+            await sleep(0.1)
+
+    test_cancelSuppression.timeout = 0.1  # type: ignore[attr-defined]
+
+    class MyError(Exception):
+        pass
+
+    async def test_cancelReplacement(self):
+        async def sleep(delay):
+            await task.deferLater(clock=reactor, delay=delay)
+
+        try:
+            await sleep(0.2)
+        except defer.CancelledError:
+            await sleep(0.1)
+            raise self.MyError("replaced error")
```

Comment:
(Again why do we want to get this from `self`?)

Reply:
Where else would I get it from? It's idiomatic to use self, but I can pull it from the class directly instead:

```suggestion
            raise __class__.MyError("replaced error")
```